test/l2/l2GraphTokenGateway.test.ts

```diff
@@ -0,0 +1,364 @@
+        tx = l2GraphTokenGateway.connect(governor.signer).setPaused(true)
+        await expect(tx).emit(l2GraphTokenGateway, 'PauseChanged').withArgs(true)
+        await expect(await l2GraphTokenGateway.paused()).eq(true)
+      })
+      describe('setPauseGuardian', function () {
+        it('cannot be called by someone other than governor', async function () {
+          const tx = l2GraphTokenGateway.connect(tokenSender.signer).setPauseGuardian(pauseGuardian.address)
+          await expect(tx).revertedWith('Caller must be Controller governor')
+        })
+        it('sets a new pause guardian', async function () {
+          const tx = l2GraphTokenGateway.connect(governor.signer).setPauseGuardian(pauseGuardian.address)
+          await expect(tx).emit(l2GraphTokenGateway, 'NewPauseGuardian').withArgs(AddressZero, pauseGuardian.address)
+        })
+        it('allows a pause guardian to pause and unpause', async function () {
+          await l2GraphTokenGateway.connect(governor.signer).setPauseGuardian(pauseGuardian.address)
+          let tx = l2GraphTokenGateway.connect(pauseGuardian.signer).setPaused(false)
+          await expect(tx).emit(l2GraphTokenGateway, 'PauseChanged').withArgs(false)
+          await expect(await l2GraphTokenGateway.paused()).eq(false)
+          tx = l2GraphTokenGateway.connect(pauseGuardian.signer).setPaused(true)
+          await expect(tx).emit(l2GraphTokenGateway, 'PauseChanged').withArgs(true)
+          await expect(await l2GraphTokenGateway.paused()).eq(true)
+        })
+      })
+    })
+  })
+
+  context('> after configuring and unpausing', function () {
+
+    const testValidOutboundTransfer = async function(signer: Signer, data: string) {
+      const tx = l2GraphTokenGateway.connect(signer)["outboundTransfer(address,address,uint256,bytes)"](
+        mockL1GRT.address,
+        l1Receiver.address,
+        toGRT('10'),
+        data
+      )
+      const expectedId = 1
+      await expect(tx).emit(l2GraphTokenGateway, 'WithdrawalInitiated')
+        .withArgs(mockL1GRT.address, tokenSender.address, l1Receiver.address, expectedId, 0, toGRT('10'))
+
+      // Should use the L1 Gateway's interface, but both come from ITokenGateway
+      const calldata = l2GraphTokenGateway.interface.encodeFunctionData('finalizeInboundTransfer',
+        [
+          mockL1GRT.address,
+          tokenSender.address,
+          l1Receiver.address,
+          toGRT('10'),
+          utils.defaultAbiCoder.encode(['uint256', 'bytes'], [0, []])
+        ]
+      )
+      await expect(tx).emit(l2GraphTokenGateway, 'TxToL1')
+        .withArgs(tokenSender.address, mockL1Gateway.address, 1, calldata)
+
+      // For some reason the call count doesn't work properly,
+      // and each function call is counted 12 times.
+      // Possibly related to https://github.com/defi-wonderland/smock/issues/85 ?
+      //expect(arbSysMock.sendTxToL1).to.have.been.calledOnce
```

Comment:
I was trying to investigate why this doesn't work, as it would be nice to explicitly check we only call this once...

Reply:
The error looks very similar to what they reported, and the number being 12 looks really fixed. I haven't used that feature from waffle yet.